Bottom: 00102e3fef60897df319f54c65eb454f26ee53bf
Top:    19ecdc3ddbe46039100f7f48adb23e95aebf752b
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-12-08 11:33:25 +0800

Refresh of tuntap-fix-possible-deadlock

---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index f7ccd79..8d85163 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -2050,11 +2050,11 @@ static int __tun_set_steering_ebpf(struct tun_struct *tun,
 		new->prog = prog;
 	}
 
-	spin_lock(&tun->lock);
+	spin_lock_bh(&tun->lock);
 	old = rcu_dereference_protected(tun->steering_prog,
-					lock_is_held(&tun->lock));
+					lockdep_is_held(&tun->lock));
 	rcu_assign_pointer(tun->steering_prog, new);
-	spin_unlock(&tun->lock);
+	spin_unlock_bh(&tun->lock);
 
 	if (old)
 		call_rcu(&old->rcu, tun_steering_prog_free);
